src/phoenix/trace/trace_dataset.py

```diff
@@ -37,6 +54,38 @@ def normalize_dataframe(dataframe: DataFrame) -> "DataFrame":
     return dataframe
 
 
+def _delete_empty_document_metadata(documents: Any) -> Any:
+    """
+    Adds a dummy field to the document metadata to make the object serializable over parquet
+    """
+    # If the documents is a list, iterate over them, check that the metadata is
+    # a dict, see if it is empty, and if it's empty, delete the metadata
+    if isinstance(documents, list):
+        # Make a deep copy of the documents
+        documents = copy.deepcopy(documents)
+        for document in documents:
+            metadata = document.get(DOCUMENT_METADATA)
+            if isinstance(metadata, dict) and not metadata:
+                # Delete the metadata object since empty dicts are not serializable
+                del document[DOCUMENT_METADATA]
+    return documents
+
+
+def get_serializable_spans_dataframe(dataframe: DataFrame) -> DataFrame:
+    """
+    Returns a dataframe that can be serialized to parquet. This means that
+    the dataframe must not contain any unserializable objects. This function
+    will delete any unserializable objects from the dataframe.
+    """
+    # Check if the dataframe has any document columns
```

Comment:
We should make a copy here, because we'll modify the columns (and to carry on pandas's functional tradition)
```suggestion
    dataframe = dataframe.copy()
    # Check if the dataframe has any document columns
```

Reply:
I always don't really know how expensive this is but will go the safe route - agree

Reply:
It's very cheap because it's just a shallow copy that copies all the pointers to the underlying numpy arrays (i.e. the columns) that make up the real data.

Reply:
ok I was not quite right. It needs to have `deep=False`.

<img width="667" alt="Screenshot 2023-12-04 at 9 56 57 AM" src="https://github.com/Arize-ai/phoenix/assets/80478925/cbe70e23-9028-40ee-a8c8-f8f9cc325a6a">

Reply:
Ah good to know - will switch